cli/tests/common/test_environment.rs

```diff
@@ -255,6 +258,74 @@ impl TestEnvironment {
     }
 }
 
+/// Helper to execute `jj` or file operation in sub directory.
+pub struct TestWorkDir<'a> {
+    env: &'a TestEnvironment,
+    root: PathBuf,
+}
+
+impl TestWorkDir<'_> {
```

Comment:
Do you imagine that we'll eventually remove `run_jj()` etc on `TestEnvironment` itself require running all commands via a `TestWorkDir`? I suppose the existing uses of `test_env.run_jj("foo");` can now be written as `let root_work_dir = test_env.work_dir(""); root_work_dir.run_jj("foo");`

Reply:
That could be, but I'm not sure. If there aren't many instances of `.run_jj_in()` (except for the workspace initialization), it might make sense to remove these.

Reply:
Actually, even if we keep the methods on `TestEnvironment`, maybe we can make them just convenience methods for `.root_work_dir().<method>`?

Reply:
That sounds like a good idea. I think we can decide it later.

Reply:
Yes, don't let this comment block